OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs

```diff
@@ -93,7 +93,8 @@ public override Activity Tick(Actor self)
 				if (self.IsDisabled())
 					return this;
 
-				var weapon = attack.ChooseArmamentsForTarget(target, forceAttack).FirstOrDefault();
+				Armament weapon = null;
+				var attack = attackArray.FirstOrDefault(af => !af.IsTraitDisabled && (weapon = af.ChooseArmamentsForTarget(target, forceAttack).FirstOrDefault()) != null);
```

Comment:
You can pull this out to a method that enumerates using a `foreach` and returns either a `Pair` of trait and weapon, returns one and uses an out param for the other, or uses out params for both and returns a bool (false if there are no matches).